pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -752,6 +741,14 @@ func (c *Reconciler) runNextSchedulableTask(ctx context.Context, pr *v1beta1.Pip
 		if rpt.IsFinalTask(pipelineRunFacts) {
 			c.setFinallyStartedTimeIfNeeded(pr, pipelineRunFacts)
 		}
+		// Check for Missing Result References
+		err := rpt.CheckMissingResultReferences(pipelineRunFacts.State)
```

Comment:
1. The `ResolveResultRefs` also calls `validateArrayResultsIndex` but `CheckMissingResultReferences` doesn't? Wonder if this will make any differences. I think what we changed in this PR should be a refactoring PR and not introduce functional changes. 
2. I wonder if we could call `CheckMissingResultReferences` in the removed line 720? Is there a reason we must call it here?

Reply:
The `ResolveResultRefs()` will have already been called at this point so it will have already validated ArrayResultsIndex(). The reason why we must call CheckMissingResultReferences() here is because when we originally call ResolveResultRefs() inside ResolvePipelineTask(), we don't return any errors here since we want to resolve all the result references at once, regardless if any of them are missing. The reason we don't want to return an error here is because an error would fail the entire pipelinerun and there are instances where we have a Task that has missing result references that will later be skipped, in which case this is a valid and we do not want to fail the pipelinerun. 

However, there is a use case that we get to `RunNextScheduableTask()` and have missing result references, that is truly a failure. This is the case if task fails to produce a result but has OnError: continue - See TestMissingResultWhenStepErrorIsIgnored().   In this case, we do need to return an error here that bubbles up to fail the PipelineRun. This is described in more detail in the PR description.  Without CheckMissingResultReferences(), TestMissingResultWhenStepErrorIsIgnored() will fail.

Let me know if that makes sense.

Reply:
The ResolveResultRefs is removed in this function and moved to ResolvePipelineTask where we don't check the error. I wonder where it is called?

I'm thinking about calling the validation function in line 720 of this file. That is the place we capture the case in TestMissingResultWhenStepErrorIsIgnored

Reply:
ResolveResultRefs() is now called inside of ResolvePipelineTask().
https://github.com/tektoncd/pipeline/blob/6df0ea18c9590880dff8311eab7b6f909d46787d/pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go#L588

I moved the `CheckMissingResultReferences()` to Line 720 to "fail faster"